package/libs/libevent2/Makefile

```diff
@@ -108,22 +106,25 @@ define Package/libevent2-pthreads/description
 	threading & locking.
 endef
 
-TARGET_CFLAGS += $(FPIC)
-
-CONFIGURE_ARGS += \
-	--enable-shared \
-	--enable-static \
-	--disable-debug-mode
+TARGET_CFLAGS += $(FPIC) -ffunction-sections -fdata-sections -flto
+TARGET_LDFLAGS += -Wl,--gc-sections,--as-needed -flto
 
-MAKE_FLAGS += \
-	CFLAGS="$(TARGET_CFLAGS)"
+CMAKE_OPTIONS += \
+	-DEVENT__DISABLE_BENCHMARK:BOOL=ON \
+	-DEVENT__DISABLE_DEBUG_MODE:BOOL=ON \
+	-DEVENT__DISABLE_REGRESS:BOOL=ON \
+	-DEVENT__DISABLE_SAMPLES:BOOL=ON \
+	$(if $(CONFIG_PACKAGE_libevent2-openssl),-DEVENT__DISABLE_OPENSSL:BOOL=OFF,-DEVENT__DISABLE_OPENSSL:BOOL=ON) \
+	$(if $(CONFIG_PACKAGE_libevent2-pthreads),-DEVENT__DISABLE_THREAD_SUPPORT:BOOL=OFF,-DEVENT__DISABLE_THREAD_SUPPORT:BOOL=ON) \
```

Comment:
If you use conditional configure options, add the variables to `PKG_CONFIG_DEPENDS`
Ref: openwrt/packages@41814de0fcc5

Reply:
Thanks, I'll have a look at it next week

Reply:
Hi, sorry I completely forgot about this!

I'm not entirely sure what you're suggesting here, 
```
PKG_CONFIG_DEPENDS := CONFIG_PACKAGE_libevent2-openssl \
PKG_CONFIG_PACKAGE_libevent2-mbedtls
```
?

Reply:
Yes